### ISSUE TITLE
Optimise Dockerfile for AWS Lambda custom runtime

### ISSUE THREAD

**Issue by alvarosanchez** [micronaut-projects/micronaut-starter#234]
Currently, the fat JAR is generated inside a container:

```
FROM gradle:6.3.0-jdk11 as builder
COPY --chown=gradle:gradle . /home/application
WORKDIR /home/application
RUN gradle clean shadowJar --no-daemon
```

But it should be faster to do it from the host, and then copy everything to the container. This is also necessary if the users wants to use `mavenLocal()`, since that wouldn't be resolvable from within the container.

Also, it should use the JDK 8/11 GraalVM versions depending on the user-selected JDK.

**Comment by graemerocher** [comment]
if we do that then it won't be possible to build images on a CI

**Comment by alvarosanchez** [comment]
Why so? We already provide a `docker-build.sh` script that could be changed to do a `./gradlew clean shadowJar` before.

**Comment by ilopmar** [comment]
@alvarosanchez keep in mind that the script should be adapted to build the fatjar with Maven too.

**Comment by alvarosanchez** [comment]
```
@if (buildTool.equals(BuildTool.GRADLE)) {
./gradlew clean shadowJar
} else {
/mvnw clean package
}
```

😉

**Comment by alvarosanchez** [comment]
Fixed via #546
